src/modules/peek/Peek.FilePreviewer/Previewers/ShellPreviewHandlerPreviewer/ShellPreviewHandlerPreviewer.cs

```diff
@@ -0,0 +1,185 @@
+        public void Dispose()
+        {
+            Clear();
+            GC.SuppressFinalize(this);
+        }
+
+        public async Task CopyAsync()
+        {
+            await Dispatcher.RunOnUiThread(async () =>
+            {
+                var storageItem = await FileItem.GetStorageItemAsync();
+                ClipboardHelper.SaveToClipboard(storageItem);
+            });
+        }
+
+        public Task<PreviewSize> GetPreviewSizeAsync(CancellationToken cancellationToken)
+        {
+            return Task.FromResult(new PreviewSize { MonitorSize = null });
+        }
+
+        public async Task LoadPreviewAsync(CancellationToken cancellationToken)
+        {
+            Clear();
+            State = PreviewState.Loading;
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Create the preview handler
+            var previewHandler = await Task.Run(() =>
+            {
+                var previewHandlerGuid = GetPreviewHandlerGuid(FileItem.Extension);
+                if (!string.IsNullOrEmpty(previewHandlerGuid))
+                {
+                    return Activator.CreateInstance(Type.GetTypeFromCLSID(Guid.Parse(previewHandlerGuid))!) as IPreviewHandler;
+                }
+                else
+                {
+                    return null;
+                }
+            });
+
+            if (previewHandler == null)
+            {
+                State = PreviewState.Error;
+                return;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Initialize the preview handler with the selected file
+            bool success = await Task.Run(() =>
+            {
+                const uint STGM_READ = 0x00000000;
+                if (previewHandler is IInitializeWithStream initWithStream)
+                {
+                    fileStream = File.OpenRead(FileItem.Path);
+                    initWithStream.Initialize(new IStreamWrapper(fileStream), STGM_READ);
+                }
+                else if (previewHandler is IInitializeWithFile initWithFile)
+                {
+                    unsafe
+                    {
+                        fixed (char* pPath = FileItem.Path)
+                        {
+                            initWithFile.Initialize(pPath, STGM_READ);
+                        }
+                    }
+                }
+                else
+                {
+                    // Handler is missing the required interfaces
+                    return false;
+                }
+
+                return true;
+            });
+
+            if (!success)
+            {
+                State = PreviewState.Error;
+                return;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // Preview.SetWindow() needs to be set in the control
+            Preview = previewHandler;
+            State = PreviewState.Loaded;
+        }
+
+        public void Clear()
```

Comment:
I'm wondering if instead of an explicit defined public Clear we could use the IPreviewHandler generated partial method PreviewChanged/OnPreviewChanged to make that unload in there.

Reply:
That won't work because a new previewer object is created whenever Peek gets called.